glossary.ftd

```diff
@@ -1,22 +1,219 @@
+processor.
+
+
+-- ft.h2: document-id
+
+The `document-id` is the url read-only property of the document interface
+returns the location from where the compiled version of document can be
+accessed. Though it can differ from [`document-full-id`](glossary/#document-full-id)
+in the respect that KVP.
+
+For a document `foo.ftd`, the `document-id` is `/foo/`
+
+This can accessed using [`document-id`](processors/document-id/)
+processor.
+
+
+-- ft.h2: document-metadata
+
+The `document-metadata` is the key-value pair data provided to
+[`document-full-id`](glossary/#document-full-id) in the
+[sitemap](glossary/#sitemap). This data can be accessed by the document, it's
+pointing to via `document-id`, using [`get-data`](processors/get-data/)
+processor.
+
+
+-- ft.code: `document-metadata` in sitemap
+lang: ftd
+
+# Foo title: /foo/
+  name: Arpita
+
+
+-- ft.markdown:
+
+Here, `/foo/` is both `document-full-id` and [`document-id`](glossary/#document-id).
+`name: Arpita` is `document-metadata` where `name` is key and `Arpita` is value.
+`foo.ftd` is the [`document-filename`](glossary/#document-filename). This
+document, `foo.ftd`, can access the `document-metadata` using `get-data`
+processor.
+
+
+
+-- ft.h2: document-suffix
+
+The `document-suffix` is the special read-only property in
+[`document-full-id`](glossary/#document-full-id) which is added after special
+character `/-/`.
+
+It helps to provide different [`document-metadata`](glossary/#document-metadata)
+to render different content.
+
+For `/foo/-/x/` as `document-full-id`, /x/ is `document-suffix`.
+
+This can accessed using [`document-suffix`](processors/document-suffix/)
+processor.
+
+
+-- ft.h2: file
+
+The unit of storage in FPM. It is uniquely identified by the
+[filename](glossary/#filename).
+
+
+
+-- ft.h2: filename
+
+The unique identifier of an [file](glossary/#file).
+
+
+
+-- ft.h2: fpm project
+
+The unit folder that contains `FPM.ftd` and any number of folders or files of
+any extension along with states (server-state and/or client-state).
+
+
+-- ft.h2: server project
+
+An fpm project which is used to track the same project but resides somewhere
+else. To communicate with server, see [clone](glossary/#clone) or
+[sync](glossary/#sync).
+
+
+
+-- ft.h2: server-latest
+
+The current tree of files in server. This can be derived by looking at the server
+history and get the latest version of each file. The file having the content
+from its latest version, if not deleted, becomes the part of current tree.
+
+
+-- ft.h2: sync
+
+To bring the contents of [client project](glossary/#client-project) into
+the [server project](glossary/#server-project). Also to bring the server's
+changes after the client diverged into the client project. In the end results
+the server and the client to be exact copy of each other.
+
+Sync is specific to a file. Though the [`fpm sync`](cli/sync/) commands
+results in syncing of all the changed file.
```

Comment:
Just delete this paragraph. I am not sure if its what information it is providing, and if the confusion it causes is worth that information.